packages/server/postgres/migrations/2025-03-18T15:55:35.723Z_normalizeUserTms.ts

```diff
@@ -0,0 +1,26 @@
+import {sql, type Kysely} from 'kysely'
+
+// biome-ignore lint/suspicious/noExplicitAny: `any` is required here since migrations should be frozen in time. alternatively, keep a "snapshot" db interface.
+export async function up(db: Kysely<any>): Promise<void> {
+  await db.schema.alterTable('User').dropColumn('tms').execute()
+}
+
+// biome-ignore lint/suspicious/noExplicitAny: `any` is required here since migrations should be frozen in time. alternatively, keep a "snapshot" db interface.
+export async function down(db: Kysely<any>): Promise<void> {
+  await db.schema
+    .alterTable('User')
+    .addColumn('tms', sql`character varying(100)[]`, (col) => col.defaultTo('{}').notNull())
+    .execute()
+  await db
+    .updateTable('User')
+    .set((eb) => ({
+      tms: eb.fn.coalesce(
+        eb
+          .selectFrom('TeamMember')
+          .select((se) => se.fn('array_agg', ['teamId']).as('tms'))
+          .where('TeamMember.userId', '=', eb.ref('User.id')),
```

Comment:
should ignore inactive teams and teamMember objects that have been removed from the team

Reply:
That archived team really got me. It's not enough to check the `TeamMember` but we also need to check the `Team` the team member points to :(